### PR TITLE
Replace PNGs with vectors

### LINKED ISSUE #375: Replace PNG files with XML ones
**Description**

Currently, there are a lot of `png` files which contribute to increase in `apk` size. Replacing possible ones with `xml` files would be better approach towards saving space and backward compatibility.

**Would like to work on this issue?**
Yes.

@mhutti1 Please share your suggestions.

### PR DESCRIPTION
**Pull request by abdulwd**
Fixes #375 

I have replaced PNGs with vector images, removed unused images and refactored a bit of code to use vectors instead of PNGs. This will reduce the size of APK drastically.

### CODE REVIEW

**Comment by brijeshshah13** [comment]
```
java.lang.RuntimeException: Unable to start activity ComponentInfo{org.kiwix.kiwixmobile/org.kiwix.kiwixmobile.KiwixMobileActivity}: android.view.InflateException: Binary XML file line #0: Error inflating class ImageButton
                                                                         at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2059)
                                                                         at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2084)
                                                                         at android.app.ActivityThread.access$600(ActivityThread.java:130)
                                                                         at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1195)
                                                                         at android.os.Handler.dispatchMessage(Handler.java:99)
                                                                         at android.os.Looper.loop(Looper.java:137)
                                                                         at android.app.ActivityThread.main(ActivityThread.java:4745)
                                                                         at java.lang.reflect.Method.invokeNative(Native Method)
                                                                         at java.lang.reflect.Method.invoke(Method.java:511)
                                                                         at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:786)
                                                                         at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:553)
                                                                         at dalvik.system.NativeStart.main(Native Method)
                                                                      Caused by: android.view.InflateException: Binary XML file line #0: Error inflating class ImageButton
                                                                         at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:704)
                                                                         at android.view.LayoutInflater.rInflate(LayoutInflater.java:746)
                                                                         at android.view.LayoutInflater.rInflate(LayoutInflater.java:749)
                                                                         at android.view.LayoutInflater.rInflate(LayoutInflater.java:749)
                                                                         at android.view.LayoutInflater.rInflate(LayoutInflater.java:749)
                                                                         at android.view.LayoutInflater.rInflate(LayoutInflater.java:749)
                                                                         at android.view.LayoutInflater.inflate(LayoutInflater.java:489)
                                                                         at android.view.LayoutInflater.inflate(LayoutInflater.java:396)
                                                                         at android.view.LayoutInflater.inflate(LayoutInflater.java:352)
                                                                         at android.support.v7.app.AppCompatDelegateImplV9.setContentView(AppCompatDelegateImplV9.java:287)
                                                                         at android.support.v7.app.AppCompatActivity.setContentView(AppCompatActivity.java:139)
                                                                         at org.kiwix.kiwixmobile.KiwixMobileActivity.onCreate(KiwixMobileActivity.java:369)
                                                                         at android.app.Activity.performCreate(Activity.java:5008)
                                                                         at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1079)
                                                                         at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2023)
                                                                         at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2084) 
                                                                         at android.app.ActivityThread.access$600(ActivityThread.java:130) 
                                                                         at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1195) 
                                                                         at android.os.Handler.dispatchMessage(Handler.java:99) 
                                                                         at android.os.Looper.loop(Looper.java:137) 
                                                                         at android.app.ActivityThread.main(ActivityThread.java:4745) 
                                                                         at java.lang.reflect.Method.invokeNative(Native Method) 
                                                                         at java.lang.reflect.Method.invoke(Method.java:511) 
                                                                         at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:786) 
                                                                         at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:553) 
                                                                         at dalvik.system.NativeStart.main(Native Method) 
                                                                      Caused by: android.content.res.Resources$NotFoundException: File res/drawable/fullscreen_exit.xml from drawable resource ID #0x7f08006d
                                                                         at android.content.res.Resources.loadDrawable(Resources.java:1923)
                                                                         at android.content.res.TypedArray.getDrawable(TypedArray.java:601)
                                                                         at android.widget.ImageView.<init>(ImageView.java:120)
                                                                         at android.widget.ImageButton.<init>(ImageButton.java:87)
                                                                         at android.support.v7.widget.AppCompatImageButton.<init>(AppCompatImageButton.java:72)
                                                                         at android.support.v7.widget.AppCompatImageButton.<init>(AppCompatImageButton.java:68)
                                                                         at android.support.v7.app.AppCompatViewInflater.createView(AppCompatViewInflater.java:118)
                                                                         at android.support.v7.app.AppCompatDelegateImplV9.createView(AppCompatDelegateImplV9.java:1016)
                                                                         at android.support.v7.app.AppCompatDelegateImplV9.onCreateView(AppCompatDelegateImplV9.java:1073)
                                                                         at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:675)
                                                                         at android.view.LayoutInflater.rInflate(LayoutInflater.java:746) 
                                                                         at android.view.LayoutInflater.rInflate(LayoutInflater.java:749) 
                                                                         at android.view.LayoutInflater.rInflate(LayoutInflater.java:749) 
                                                                         at android.view.LayoutInflater.rInflate(LayoutInflater.java:749) 
                                                                         at android.view.LayoutInflater.rInflate(LayoutInflater.java:749) 
                                                                         at android.view.LayoutInflater.inflate(LayoutInflater.java:489) 
                                                                         at android.view.LayoutInflater.inflate(LayoutInflater.java:396) 
                                                                         at android.view.LayoutInflater.inflate(LayoutInflater.java:352) 
                                                                         at android.support.v7.app.AppCompatDelegateImplV9.setContentView(AppCompatDelegateImplV9.java:287) 
                                                                         at android.support.v7.app.AppCompatActivity.setContentView(AppCompatActivity.java:139) 
                                                                         at org.kiwix.kiwixmobile.KiwixMobileActivity.onCreate(KiwixMobileActivity.java:369) 
                                                                         at android.app.Activity.performCreate(Activity.java:5008) 
                                                                         at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1079) 
                                                                         at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2023) 
                                                                         at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2084) 
                                                                         at android.app.ActivityThread.access$600(ActivityThread.java:130) 
                                                                         at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1195) 
                                                                         at android.os.Handler.dispatchMessage(Handler.java:99) 
                                                                         at android.os.Looper.loop(Looper.java:137) 
                                                                         at android.app.ActivityThread.main(ActivityThread.java:4745) 
                                                                         at java.lang.reflect.Method.invokeNative(Native Method) 
                                                                         at java.lang.reflect.Method.invoke(Method.java:511) 
                                                                         at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:786) 
                                                                         at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:553) 
                                                                         at dalvik.system.NativeStart.main(Native Method) 
                                                                      Caused by: org.xmlpull.v1.XmlPullParserException: Binary XML file line #0: invalid drawable tag vector
                                                                         at android.graphics.drawable.Drawable.createFromXmlInner(Drawable.java:877)
                                                                         at android.graphics.drawable.Drawable.createFromXml(Drawable.java:818)
                                                                         at android.content.res.Resources.loadDrawable(Resources.java:1920)
                                                                         at android.content.res.TypedArray.getDrawable(TypedArray.java:601) 
                                                                         at android.widget.ImageView.<init>(ImageView.java:120) 
                                                                         at android.widget.ImageButton.<init>(ImageButton.java:87) 
                                                                         at android.support.v7.widget.AppCompatImageButton.<init>(AppCompatImageButton.java:72) 
                                                                         at android.support.v7.widget.AppCompatImageButton.<init>(AppCompatImageButton.java:68) 
                                                                         at android.support.v7.app.AppCompatViewInflater.createView(AppCompatViewInflater.java:118) 
                                                                         at android.support.v7.app.AppCompatDelegateImplV9.createView(AppCompatDelegateImplV9.java:1016) 
                                                                         at android.support.v7.app.AppCompatDelegateImplV9.onCreateView(AppCompatDelegateImplV9.java:1073) 
                                                                         at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:675) 
                                                                         at android.view.LayoutInflater.rInflate(LayoutInflater.java:746) 
                                                                         at android.view.LayoutInflater.rInflate(LayoutInflater.java:749) 
                                                                         at android.view.LayoutInflater.rInflate(LayoutInflater.java:749) 
                                                                         at android.view.LayoutInflater.rInflate(LayoutInflater.java:749) 
                                                                         at android.view.LayoutInflater.rInflate(LayoutInflater.java:749) 
                                                                         at android.view.LayoutInflater.inflate(LayoutInflater.java:489) 
                                                                         at android.view.LayoutInflater.inflate(LayoutInflater.java:396) 
                                                                         at android.view.LayoutInflater.inflate(LayoutInflater.java:352) 
                                                                         at android.support.v7.app.AppCompatDelegateImplV9.setContentView(AppCompatDelegateImplV9.java:287) 
                                                                         at android.support.v7.app.AppCompatActivity.setContentView(AppCompatActivity.java:139) 
                                                                         at org.kiwix.kiwixmobile.KiwixMobileActivity.onCreate(KiwixMobileActivity.java:369) 
                                                                         at android.app.Activity.performCreate(Activity.java:5008) 
                                                                         at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1079) 
                                                                         at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2023) 
                                                                         at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2084) 
                                                                         at android.app.ActivityThread.access$600(ActivityThread.java:130) 
                                                                         at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1195) 
                                                                         at android.os.Handler.dispatchMessage(Handler.java:99) 
                                                                         at android.os.Looper.loop(Looper.java:137) 
                                                                         at android.app.ActivityThread.main(ActivityThread.java:4745) 
                                                                         at java.lang.reflect.Method.invokeNative(Native Method) 
                                                                         at java.lang.reflect.Method.invoke(Method.java:511) 
                                                                         at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:786) 
                                                                         at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:553) 
                                                                         at dalvik.system.NativeStart.main(Native Method) 

```
on sdk 16 emulator

**Comment by abdulwd** [comment]
@brijeshshah13 Thanks! For bringing this to light. Now you may test it again. I forgot to change some lines. My bad.
@mhutti1 @kelson42 Now it's fixed. I have tested it on a virtual device running API 14. I have removed 96 PNGs but I have kept back some PNGs, reasons for which include back compatibility or those PNGs are too complicated to be converted to vectors and for them, the rasterized version will be faster than the vectorized version. Removing those will do more harm than good. After the merge of this PR, the APK size will reduce by a considerable amount. :)

File: app/src/main/res/drawable/fullscreen_exit.xml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+        android:width="24dp"
```
Comment (brijeshshah13) [inline app/src/main/res/drawable/fullscreen_exit.xml]: Please increase the height and width of the `fullscreen_exit` drawable to `36dp`. Current icon can hardly be noticed.

Comment (abdulwd) [reply app/src/main/res/drawable/fullscreen_exit.xml]: Missed it. Fixed. Thanks again. :)

**Comment by brijeshshah13** [comment]
@mhutti1 Thank you for pointing that out. I'm also getting this in the log:

E/AndroidRuntime: FATAL EXCEPTION: main
                                                                     android.app.RemoteServiceException: Bad notification posted from package org.kiwix.kiwixmobile: Couldn't expand RemoteViews for: StatusBarNotification(pkg=org.kiwix.kiwixmobile id=-1510021710 tag=null score=0 notn=Notification(pri=0 contentView=org.kiwix.kiwixmobile/0x1090071 vibrate=null sound=null defaults=0x0 flags=0x2 kind=[null] 2 actions))
                                                                         at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1361)
                                                                         at android.os.Handler.dispatchMessage(Handler.java:99)
                                                                         at android.os.Looper.loop(Looper.java:137)
                                                                         at android.app.ActivityThread.main(ActivityThread.java:4745)
                                                                         at java.lang.reflect.Method.invokeNative(Native Method)
                                                                         at java.lang.reflect.Method.invoke(Method.java:511)
                                                                         at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:786)
                                                                         at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:553)
                                                                         at dalvik.system.NativeStart.main(Native Method)

**Comment by abdulwd** [comment]
I was testing it on an emulator and it's quite slow, therefore, I did not download a file. I'll fix it. :)

**Comment by mhutti1** [comment]
I think theres an issue with the network on earlier android emualulators I will make an issue.

**Comment by abdulwd** [comment]
This issue took time.
I have tested it on **API 17** and **API 24**. Everything is working fine. The crash has been fixed which involved reverting some changes to keep the app back compatible.
@mhutti1 _donwloadTest_ and _networkTest_ are failing but the same tests are failing on branch **master**. These tests in any way I don't think are failing because of the changes I have made.

**Comment by abdulwd** [comment]
@mhutti1 Please review this. :smile:

File: app/src/main/res/drawable/action_bookmark.xml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="36dp"
+    android:height="36dp"
+    android:viewportHeight="24.0"
+    android:viewportWidth="24.0">
+    <path
```
Comment (mhutti1) [inline app/src/main/res/drawable/action_bookmark.xml]: 2 space tabs for XML please in all locations.

Comment (abdulwd) [reply app/src/main/res/drawable/action_bookmark.xml]: Done!

**Comment by abdulwd** [comment]
@mhutti1 I have changed the indentation. I hope it gets merged this time as this issue has been opened for long. :)

**Pull request merged**